lib/pg_online_schema_change/orchestrate.rb

```diff
@@ -239,6 +239,7 @@ def swap!
         Replay.play!(rows, opened)
 
         sql = <<~SQL
+          SELECT setval((select pg_get_serial_sequence(\'#{shadow_table}\', \'#{primary_key}\')), (SELECT max(#{primary_key}) FROM #{client.table})+1);
```

Comment:
doing from here also since there is an exclusive lock already acquired on the original/primary table.

Reply:
NB: Another option would be to grab `currval()` of the original sequence.

Reply:
Yeah, thats what I had at first, but the test case was a bit off since it kept throwing sequence not registered or similar. I think using currval will mean that we won’t have to worry about type (string, etc) either right ?

i will look into restructuring the tests tomorrow

Reply:
I think if it is anything other than an integer type, there won't be a sequence attached. Instead there will either be a default function like `uuid_generate_v1()` or nothing because it's being generated by the application. So, if we wanted to support other types, but still properly set a sequence value if it exists, we just need to check if there's an attached sequence.

Reply:
doh! yes, I think we can just to a sequence check. Nice, will update in a follow up PR.